be/src/exec/pipeline/query_context.cpp

```diff
@@ -244,4 +244,20 @@ void QueryContextManager::clear() {
     _context_maps.clear();
 }
 
+bool QueryContextManager::is_overloaded(int degree_of_parallelism) {
+    static const int max_num_tasks =
```

Comment:
Shall we check the overload by the `pipeline_exec_thread_pool_thread_num` instead of CPU numbers?

Reply:
And at the same time, shall we decrease the queue size of  `_driver_executor`  ? Or set it according to this task limit?

Reply:
> Shall we check the overload by the `pipeline_exec_thread_pool_thread_num` instead of CPU numbers?

OK, great.

Reply:
> And at the same time, shall we decrease the queue size of `_driver_executor` ? Or set it according to this task limit?

The queue size of `_driver_executor` is unlimited for now. I think limiting the driver queue size after decomposing to pipeline drivers and limiting the estimated tasks before decomposing to pipeline drivers are two different refusal strategies.